subprojects/docs/src/docs/userguide/workingWithFiles.adoc

```diff
@@ -15,47 +15,288 @@
+
+So how do you ensure that `reports` itself is copied across, but not any other directory in `$buildDir`? The answer is to add it as an include pattern:
+
+----
+task copyReports(type: Copy) {
+    from("${buildDir}") {
+        include "reports/**"
+    }
+    into "${buildDir}/toArchive"
+}
+----
+
+You'll get the same behavior as before except with one extra level of directory in the destination, i.e. `toArchive/reports`.
+
+One thing to note is how the `include()` directive applies only to the `from()`, whereas the directive in the previous section applied to the whole task. These different levels of granularity in the copy specification allow you to easily handle most requirements that you will come across. TODO: Link to more details on copy specs.
+
+=== Creating archives (zip, tar, etc.)
+
+From the perspective of Gradle, packing files into a single archive is effectively a copy in which the destination is an archive file instead of a directory on the file system. This means that archiving files looks a lot like copying, with all of the same features!
+
+The simplest case involves packing all the contents of a directory into an archive, such as a zip in this example:
+
+----
+task packageDistribution(type: Zip) {
+    archiveName = "my-distribution.zip"
+    destinationDir = file("${buildDir}/dist")
+
+    from "${buildDir}/toArchive"
+}
+----
+
+Each type of archive has its own task type, the most common ones being api:org.gradle.api.tasks.bundling.Zip[], api:org.gradle.api.tasks.bundling.Tar[] and api:org.gradle.api.tasks.bundling.Jar[]. And instead of having an `into()` setting like `Copy`, you configure the location and name of the destination archive file. In many cases you don't even need to do that as `destinationDir` defaults to `${buildDir}/distributions` and many plugins provide conventions for `archiveName`.
+
+The archiving tasks share most of the configuration options of `Copy`, including filtering and renaming. One of the most common options people use allows you to create a subdirectory in the destination to copy the files into. For example, let's say you want to package all PDFs into a `docs` directory in the root of the archive. This `docs` directory doesn't exist in the source location, so you have to create it as part of the archive. You do this by adding an `into()` declaration for just the PDFs:
+
+----
+task packageDistribution(type: Zip) {
+    archiveName = "my-distribution.zip"
+    destinationDir = file("${buildDir}/dist")
+
+    from("${buildDir}/toArchive") {
+        exclude "**/*.pdf"
+    }
+
+    from("${buildDir}/toArchive") {
+        include "**/*.pdf"
+        into "docs"
+    }
+}
+----
+
+As you can see, you can have multiple `from()` declarations in a copy specification, each with its own configuration.
```

Comment:
We should have a section that explain the file operation methods on `Project` (e.g. `project.copy`) and the difference between the task types and the methods.

Reply:
That was already in the chapter, but I've updated it. You'll find it in the File Copying in Depth section. It's actually focused on the `copy()` method, but now you mention it, there isn't anything on `delete()` or its corresponding task right now. Are there any others I should cover?

Reply:
Add information on the following:

 - [ ] `Delete` task and method
 - [ ] `sync()` method